client/ayon_resolve/plugins/publish/collect_editorial_package.py

```diff
@@ -1,88 +1,56 @@
-            if (
-                data.get("publish")
-                and data["publish"].get("productType") != "editorial_pkg"
-            ):
-                continue
-
-            instance = context.create_instance(name=media_pool_item.GetName())
-
-            publish_data = data["publish"]
-
-            # get version from publish data and rise it one up
-            version = publish_data.get("version")
-            if version is not None:
-                version += 1
-
-                # make sure last version of product is higher than current
-                # expected current version from publish data
-                folder_entity = ayon_api.get_folder_by_path(
-                    project_name=project_name,
-                    folder_path=publish_data["folderPath"],
-                )
-                last_version = ayon_api.get_last_version_by_product_name(
-                    project_name=project_name,
-                    product_name=publish_data["productName"],
-                    folder_id=folder_entity["id"],
-                )
-                if last_version is not None:
-                    last_version = int(last_version["version"])
-                    if version <= last_version:
-                        version = last_version + 1
-
-                publish_data["version"] = version
-
-            publish_data.update(
-                {
-                    "mediaPoolItem": media_pool_item,
-                    "item": media_pool_item,
-                }
+            if media_pool_item.GetUniqueId() == media_pool_item_id:
+                break
+        else:
+            raise RuntimeError("Could not identify media pool item from instance.")
+
+        # get version from publish data and rise it one up
+        version = instance.data.get("version")
+        if version is not None:
+            version += 1
+
+            # make sure last version of product is higher than current
+            # expected current version from publish data
+            folder_entity = ayon_api.get_folder_by_path(
+                project_name=project_name,
+                folder_path=publish_data["folderPath"],
+            )
+            last_version = ayon_api.get_last_version_by_product_name(
+                project_name=project_name,
+                product_name=publish_data["productName"],
+                folder_id=folder_entity["id"],
             )
+            if last_version is not None:
+                last_version = int(last_version["version"])
+                if version <= last_version:
+                    version = last_version + 1
 
-            if publish_data.get("taskId"):
-                task_entity = get_task_by_id(
-                    project_name=project_name,
-                    task_id=publish_data["taskId"],
-                )
-                publish_data["taskEntity"] = task_entity
-                publish_data["task"] = task_entity["name"]
+            instance.data["version"] = version
```

Comment:
This seems redundant - an instance should always automatically pick a version that is the next version that is available?

We should be able to take this out completely?

Which means - that if this is removed, maybe this full collector becomes redundant because the media pool item is available in the transient data - and hence... it's not collecting anything of use anymore? We may just need to update any other usage of `instance.data["mediaPoolItem"` and `instance.data["item"]` for this family?

Reply:
Good call, that's something I expected @jakubjezek001 could shed some light on. 
I initially thought this could be removed because this whole logic seems trigged when provided `version` is not None and the creator does not set this up. But wanted to check with you first in case that `version` edited by some hidden mechanism..

Reply:
Yeah this workflow is looking strange but it is the client preference. Use next available version if timeline is loaded from editorial package (this way we preserve the explicit chain), if timeline is created via Creator then find next available arbitrary version.

Reply:
Update, we had a call aside and agreed to leave this one as-is for now. An explanatory comment has been added in the plugin tho.